### PR TITLE
Fix: Enable GPU + Native Attention and restore correct decode

### PR DESCRIPTION
**Pull request by qyzhaoxun**
**Related:** [#204](https://github.com/sgl-project/sglang-jax/issues/204#issuecomment-3343607087)

## Current issues on GPU
- `jnp.argsort` is unavailable
- **KV cache is not being persisted across layers**, which breaks decode

## This PR achieves
1. The inference engine now **starts and runs reliably** under **GPU(L20) + native attention**.
2. **Decode outputs are corrected** (no longer stuck on a single token or a constant token like `220`).

```
# JAX_COMPILATION_CACHE_DIR=/tmp/jit_cache python3 -u -m sgl_jax.launch_server --model-path Qwen/Qwen1.5-4B-Chat --trust-remote-code --device=cuda --xla-backend=cuda --random-seed=3 --mem-fraction-static=0.8 --max-prefill-tokens=64 --max-running-requests=1  --log-requests --log-requests-level=2 --log-level-http=debug --show-time-cost --decode-log-interval=1 --enable-request-time-stats-logging --attention-backend=native  --dtype=bfloat16 --skip-server-warmup
```
```
 curl -s http://localhost:30000/v1/chat/completions   -H "Content-Type: application/json"   -d '{
    "model": "Qwen/Qwen1.5-4B-Chat",
    "messages": [{"role": "user", "content": "10+2=? "}],
    "skip_special_tokens": true,
    "max_tokens": 10,
    "temperature": 0.0,
    "top_p": 1.0,
    "top_k": -1,
    "min_p": 0.0,
    "stream": false
  }'
{"id":"c0ef9f1fb134461ba0263e883a270a5b","object":"chat.completion","created":1760277969,"model":"Qwen/Qwen1.5-4B-Chat","choices":[{"index":0,"message":{"role":"assistant","content":"10 + 2 = 12","reasoning_content":null,"tool_calls":null},"logprobs":null,"finish_reason":"length","matched_stop":null,"hidden_states":null}],"usage":{"prompt_tokens":25,"total_tokens":35,"completion_tokens":10,"prompt_tokens_details":null}}
```
```
curl -s http://localhost:30000/v1/chat/completions   -H "Content-Type: application/json"   -d '{
    "model": "Qwen/Qwen1.5-4B-Chat",
    "messages": [{"role": "user", "content": "tell me a story: "}],
    "skip_special_tokens": false,
    "max_tokens": 100,
    "temperature": 0.7,
    "top_p": 0.9,
    "top_k": 50,
    "stream": false
  }'
{"id":"fdbe982f8ac24875bab0faebb7103209","object":"chat.completion","created":1760277993,"model":"Qwen/Qwen1.5-4B-Chat","choices":[{"index":0,"message":{"role":"assistant","content":"Once upon a time, in a far-off land, there was a young girl named Lily. Lily lived in a small village surrounded by lush green forests and sparkling streams. She was a kind-hearted and curious girl who loved exploring the world around her.\n\nOne day, while wandering through the forest, Lily stumbled upon a hidden cave. The entrance was small and dark, but she felt a sense of excitement and curiosity about what might be inside. She took a deep breath and stepped inside.\n\nAs she entered","reasoning_content":null,"tool_calls":null},"logprobs":null,"finish_reason":"length","matched_stop":null,"hidden_states":null}],"usage":{"prompt_tokens":25,"total_tokens":125,"completion_tokens":100,"prompt_tokens_details":null}}
```

### CODE REVIEW

**Review by jimoosciuc — DISMISSED** [review]
LGTM

File: python/sgl_jax/srt/layers/attention/native_backend.py [hunk 1]
```diff
@@ -163,8 +171,11 @@ def forward_attention(
     Returns:
         Output tensor of shape[batch_size, hidden_size]
     """
-    k_cache = jnp.take(k_cache, loc, axis=0)
-    v_cache = jnp.take(v_cache, loc, axis=0)
+
+    cache_size = k_cache.shape[0]
+    safe_loc = jnp.where(loc > 0, loc, cache_size)
```
[the remark this answers is not in the document]
Comment (jimoosciuc) [reply python/sgl_jax/srt/layers/attention/native_backend.py:177]: @qyzhaoxun Could please confirm the warning here? loc should be able to include 0.

[the remark this answers is not in the document]
Comment (qyzhaoxun) [reply python/sgl_jax/srt/layers/attention/native_backend.py:177]: I double-checked the logs and implementation: KV-cache padding is read as 0, which is introduced during precompilation. Valid indices begin at 1. https://github.com/sgl-project/sglang-jax/blob/main/python/sgl_jax/srt/mem_cache/allocator.py#L99
